pkg/knowledge_base2/properties/resource_property.go

```diff
@@ -0,0 +1,164 @@
+package properties
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/collectionutil"
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
+)
+
+type (
+	ResourceProperty struct {
+		DefaultValue *construct.ResourceId  `json:"default_value" yaml:"default_value"`
+		AllowedTypes construct.ResourceList `yaml:"allowed_types"`
+		Namespace    bool                   `yaml:"namespace"`
+		SharedPropertyFields
+		*knowledgebase.PropertyDetails
+	}
+)
+
+func (r *ResourceProperty) SetProperty(resource *construct.Resource, value any) error {
+	if val, ok := value.(construct.ResourceId); ok {
+		return resource.SetProperty(r.Path, val)
+	} else if val, ok := value.(construct.PropertyRef); ok {
+		return resource.SetProperty(r.Path, val)
+	}
+	return fmt.Errorf("invalid resource value %v", value)
+}
+
+func (r *ResourceProperty) AppendProperty(resource *construct.Resource, value any) error {
+	return r.SetProperty(resource, value)
+}
+
+func (r *ResourceProperty) RemoveProperty(resource *construct.Resource, value any) error {
+	delete(resource.Properties, r.Path)
```

Comment:
yeah for things like sets/lists you could remove an item

will change the method for these though like you mentioned it might not work for nested ones

Reply:
Isn't that what `resource.RemoveProperty(path, optionalKeyOrElement)` does already?

Reply:
i guess it depends on if we want validation around the value theyre passing in or anything. Likely should be fine if its all already handled